src/cocotb/regression.py

```diff
@@ -165,132 +252,75 @@ def __init__(self) -> None:
-
-    def register_test(
-        self,
-        func: Callable[..., Coroutine[Any, Any, None]],
-        name: Optional[str] = None,
-        module: Optional[str] = None,
-        timeout_time: Optional[float] = None,
-        timeout_unit: str = "step",
-        expect_fail: bool = False,
-        expect_error: Union[Type[Exception], Sequence[Type[Exception]]] = (),
-        skip: bool = False,
-        stage: int = 0,
-    ) -> None:
-        """Register a test with the RegressionManager.
+        Only those tests which match at least one of the given filters are included;
+        the rest are excluded.
 
         Should be called before :meth:`start_regression` is called.
 
         Args:
-            func:
-                The test function to register.
-
-            name:
-                The name of the test function.
-                Defaults to ``func.__qualname__`` if not given.
-
-            module:
-                The name of the module containing the test function.
-                Defaults to ``func.__module__`` if not given.
-
-            timeout_time:
-                Simulation time duration before timeout occurs.
+            filters: Each argument given is a regex pattern for test names.
+                A match *includes* the test.
+        """
+        for filter in filters:
+            compiled_filter = re.compile(filter)
+            self._filters.append(compiled_filter)
 
-            timeout_unit:
-                Units of ``timeout_time``, accepts any units that :class:`~cocotb.triggers.Timer` does.
+    def set_mode(self, mode: RegressionMode) -> None:
+        """Set the regression mode.
 
-            expect_fail:
-                If ``True`` and the test fails a functional check via an ``assert`` statement, :class:`pytest.raises`,
-                :class:`pytest.warns`, or :class:`pytest.deprecated_call`, the test is considered to have passed.
-                If ``True`` and the test passes successfully, the test is considered to have failed.
+        See :class:`RegressionMode` for more details on how each mode affects :class:`RegressionManager` behavior.
 
-            expect_error:
-                Mark the result as a pass only if one of the exception types is raised in the test.
-                This is primarily for cocotb internal regression use for when a simulator error is expected.
+        Should be called before :meth:`start_regression` is called.
+        """
+        self._mode = mode
 
-            skip:
-                Don't execute this test as part of the regression.
-                The test can still be run manually by setting :make:var:`TESTCASE`.
+    def register_test(self, test: Test) -> None:
+        """Register a test with the :class:`RegressionManager`.
 
-            stage:
-                Order tests logically into stages, where multiple tests can share a stage.
+        Should be called before :meth:`start_regression` is called.
```

Comment:
```suggestion
        Should be called before :meth:`start_regression` is called.
        
        Args:
            test: The test object to register.
```
Looks better in the generated docs, and may be helpful in some IDE.